### PR TITLE
[Communication]: Rename ConnectionString and move it to implementation

### PR DESCRIPTION
**Pull request by JianpingChen**
https://apiview.dev/Assemblies/Review/ddeb4729c4104f0bb69c8f0beaa055e5

Link above is the API review after the change

### CODE REVIEW

**Comment by JianpingChen** [comment]
Required for API review.

https://apiview.dev/Assemblies/Review/ddeb4729c4104f0bb69c8f0beaa055e5

File: sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberClientBuilder.java [hunk 1]
```diff
@@ -137,7 +136,7 @@ public PhoneNumberClientBuilder credential(TokenCredential tokenCredential) {
      */
     public PhoneNumberClientBuilder connectionString(String connectionString) {
         Objects.requireNonNull(connectionString, "'connectionString' cannot be null.");
-        ConnectionString connectionStringObject = new ConnectionString(connectionString);
+        com.azure.communication.common.implementation.CommunicationConnectionString connectionStringObject = new com.azure.communication.common.implementation.CommunicationConnectionString(connectionString);
```
Comment (DominikMe) [inline sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberClientBuilder.java:139]: nit: can we shorten this by adding another import?

File: sdk/communication/azure-communication-administration/src/test/java/com/azure/communication/administration/PhoneNumberIntegrationTestBase.java [hunk 2]
```diff
@@ -10,7 +10,6 @@
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
-import com.azure.communication.common.ConnectionString;
```
Comment (DominikMe) [inline sdk/communication/azure-communication-administration/src/test/java/com/azure/communication/administration/PhoneNumberIntegrationTestBase.java:13]: Why not 
`import com.azure.communication.common.implementation.ConnectionString;` ?

File: sdk/communication/azure-communication-administration/src/test/java/com/azure/communication/administration/PhoneNumberIntegrationTestBase.java [hunk 3]
```diff
@@ -93,16 +92,16 @@ protected PhoneNumberClientBuilder getClientBuilderWithConnectionString(HttpClie
             redactors.add(data -> redact(data, JSON_PROPERTY_VALUE_REDACTION_PATTERN.matcher(data), "REDACTED"));
             builder.addPolicy(interceptorManager.getRecordPolicy(redactors));
         }
-
```
Comment (DominikMe) [inline sdk/communication/azure-communication-administration/src/test/java/com/azure/communication/administration/PhoneNumberIntegrationTestBase.java:96]: I like this whitespace cleanup :-)

File: sdk/communication/azure-communication-common/src/main/java/com/azure/communication/common/CommunicationTokenCredential.java [hunk 4]
```diff
@@ -35,13 +35,13 @@
...
-     * 
-     * @param token serialized JWT token
+     * Create with serialized JWT initialToken
+     *
+     * @param initialToken serialized JWT initialToken
      */
-    public CommunicationTokenCredential(String token) {
-        Objects.requireNonNull(token, "'token' cannot be null.");
-        setToken(token);
+    public CommunicationTokenCredential(String initialToken) {
```
Comment (DominikMe) [inline sdk/communication/azure-communication-common/src/main/java/com/azure/communication/common/CommunicationTokenCredential.java:42]: As I remember, we had talked and decided to use `token` and not `initialToken` as the name, so please revert the change to make it match C#. Sorry ...

https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/communication/Azure.Communication.Common/src/CommunicationTokenCredential.cs

File: sdk/communication/azure-communication-common/src/main/java/com/azure/communication/common/CommunicationTokenRefreshOptions.java [hunk 5]
```diff
@@ -22,23 +22,23 @@
...
      * @param refreshProactively when set to true, turn on proactive fetching to call
-     *                           tokenRefresher before token expiry by minutes set
+     *                           tokenRefresher before initialToken expiry by minutes set
      *                           with setCallbackOffsetMinutes or default value of
      *                           two minutes
-     * @param token the optional serialized JWT token
+     * @param initialToken the optional serialized JWT initialToken
      */
-    public CommunicationTokenRefreshOptions(TokenRefresher tokenRefresher, boolean refreshProactively, String token) {
+    public CommunicationTokenRefreshOptions(TokenRefresher tokenRefresher, boolean refreshProactively, String initialToken) {
```
Comment (DominikMe) [inline sdk/communication/azure-communication-common/src/main/java/com/azure/communication/common/CommunicationTokenRefreshOptions.java:38]: Same, let's stick with `token`

File: sdk/communication/azure-communication-sms/src/main/java/com/azure/communication/sms/SmsClientBuilder.java [hunk 6]
```diff
@@ -103,7 +102,7 @@ public SmsClientBuilder credential(TokenCredential tokenCredential) {
      */
     public SmsClientBuilder connectionString(String connectionString) {
         Objects.requireNonNull(connectionString, "'connectionString' cannot be null.");
-        ConnectionString connectionStringObject = new ConnectionString(connectionString);
+        com.azure.communication.common.implementation.CommunicationConnectionString connectionStringObject = new com.azure.communication.common.implementation.CommunicationConnectionString(connectionString);
```
Comment (DominikMe) [inline sdk/communication/azure-communication-sms/src/main/java/com/azure/communication/sms/SmsClientBuilder.java:105]: nit: let's add an import at the top

File: sdk/communication/azure-communication-sms/src/test/java/com/azure/communication/sms/SmsLiveTestBase.java [hunk 7]
```diff
@@ -55,7 +54,7 @@ protected SmsClientBuilder getSmsClientBuilder(HttpClient httpClient) {
 
     protected SmsClientBuilder getSmsClientBuilderWithManagedIdentity(HttpClient httpClient) {
         SmsClientBuilder builder = new SmsClientBuilder();
-        String livetestEndpoint = new ConnectionString(CONNECTION_STRING).getEndpoint();
+        String livetestEndpoint = new com.azure.communication.common.implementation.CommunicationConnectionString(CONNECTION_STRING).getEndpoint();
```
Comment (DominikMe) [inline sdk/communication/azure-communication-sms/src/test/java/com/azure/communication/sms/SmsLiveTestBase.java:57]: nit: use import

File: sdk/communication/azure-communication-common/src/main/java/com/azure/communication/common/CommunicationClientCredential.java [hunk 8]
```diff
@@ -1,55 +1,13 @@
...
-import java.util.HashMap;
-import java.util.Locale;
-import java.util.Map;
 import java.util.Objects;
-import java.util.stream.Collectors;
 
 /**
  * Credential to support accessing communication resource with resource access key
  */
 public final class CommunicationClientCredential {
```
Comment (DominikMe) [inline sdk/communication/azure-communication-common/src/main/java/com/azure/communication/common/CommunicationClientCredential.java:9]: We can delete this whole class now and instead use `AzureKeyCredential`. We're already doing that in C# and Js.

https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/core/azure-core/src/main/java/com/azure/core/credential/AzureKeyCredential.java

**Review by DominikMe — COMMENTED** [review]
Cool stuff, added some comments

File: sdk/communication/azure-communication-common/src/main/java/com/azure/communication/common/CommunicationTokenRefreshOptions.java (hunk 5 above)
Comment (RezaJooyandeh) [reply sdk/communication/azure-communication-common/src/main/java/com/azure/communication/common/CommunicationTokenRefreshOptions.java:38]: The other `ctor` is static token so we call it `token` but this one is just the initial one, so we should call it `initialToken`

File: sdk/communication/azure-communication-common/src/main/java/com/azure/communication/common/CommunicationClientCredential.java (hunk 8 above)
Comment (JianpingChen) [reply sdk/communication/azure-communication-common/src/main/java/com/azure/communication/common/CommunicationClientCredential.java]: Deleted

File: sdk/communication/azure-communication-administration/src/main/java/module-info.java [hunk 9]
```diff
@@ -4,6 +4,7 @@
 module com.azure.communication.administration {
 
     requires transitive com.azure.communication.common;
+    requires com.azure.core;
```
Comment (JianpingChen) [inline sdk/communication/azure-communication-administration/src/main/java/module-info.java:7]: requires transitive

**Review by DominikMe — CHANGES_REQUESTED** [review]
Found one flaw in the serializer logic

File: sdk/communication/azure-communication-common/src/main/java/com/azure/communication/common/CommunicationIdentifierSerializer.java [hunk 10]
```diff
@@ -29,16 +37,29 @@ public static CommunicationIdentifier deserialize(CommunicationIdentifierModel i
...
         }
 
-        Objects.requireNonNull(id);
-        return new UnknownIdentifier(id);
+        if (kind == CommunicationIdentifierKind.UNKNOWN) {
+            Objects.requireNonNull(id);
+            return new UnknownIdentifier(id);
+        }
+
+        throw new IllegalArgumentException(String.format("Unknown identifier kind '%s'", kind));
```
[the remark this answers is not in the document]
Comment (JianpingChen) [reply sdk/communication/azure-communication-common/src/main/java/com/azure/communication/common/CommunicationIdentifierSerializer.java]: Reverted

File: sdk/communication/azure-communication-common/src/main/java/com/azure/communication/common/CommunicationIdentifierSerializer.java [hunk 11]
```diff
@@ -6,7 +6,15 @@
 import java.util.Objects;
 
 class CommunicationIdentifierSerializer {
-    public static CommunicationIdentifier deserialize(CommunicationIdentifierModel identifier) {
+    /**
+     * Deserialize CommunicationIdentifierModel into CommunicationIdentifier
+     * @param identifier CommunicationIdentifierModel to be deserialized
+     * @return deserialized CommunicationIdentifier
+     * @throws IllegalArgumentException when identifier has unknown CommunicationIdentifierKind
```
Comment (DominikMe) [inline sdk/communication/azure-communication-common/src/main/java/com/azure/communication/common/CommunicationIdentifierSerializer.java:13]: We can remove the `@throws` now

**Review by DominikMe — APPROVED** [review]
Thanks for addressing everything @JianpingChen

One nit to update a comment, besides that lgtm!
